src/components/projects-section/index.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+
+import styles from "./styles.module.css";
+import { useState } from "react";
+import { ProjectsIcon } from "./projects-icon";
+import { mock_projects_data, mock_projects_buttons } from "@/mock-data/projects";
+import { ProjectsCarousel } from "../projects-carousel";
+
+export function ProjectsSection() {
```

Comment:
Uma coisa que acho importante: aqui estamos trasnformando todo o nosso componente em client. Legal isolar ao maximo e aproveitar a caracteristica do Next de server-first. Joga os states para dentro de um componente Carousel, que vai lidar com a interatividade. O projectsSection vai ser um async server component, e nele que vai acontecer o fetch dos projetos quando estivermos pegando ele da api.

Reply:
A ideia é ótima, mesmo, mas fiquei pensando que, como gerencio os slides E os botões de cima no mesmo componente e isso é tudo que existe na seção não vi muito sentido em criar mais um nível de componente pra descer os estados

Reply:
Quando a gente for fazer o fetch vai fazer mais sentido. Eh melhor a gente fazer esse fetch no servidor, e com o useState nao podemos fazer isso. E a cada mudanca de esado ele vai rerenderizar o componente todo. Agor ao ganho de performance eh minimo, mas melhor ja deixar isso preparado pra quando os dados vierem de fora